### ISSUE TITLE
Migrate to AndroidX

### ISSUE THREAD

**Issue by hso132** [DivisionBy-Zero/erpa-sweng#136]
- Work around a bug in gradle 3.2.4

**Comment by roosemberth** [comment]
TL;DR: We can't:

- AndroidX conflicts with appcompat. We can refactor our code, but we can't refactor our libs (dependencies).
- Gradle build tools provides Jetify, which automatically recompiles class code to use the AndroidX compatibility layer to AppCompat. Jetify is still in an early stage. It works for “classic classes” but is not guaranteed to work on code with advanced functionnality such as reflection or Bytecode instrumentation.

If upstream has ported to AndroidX, it's advised to use the ported version and disable jetify.
Since some of the “simple” libs we use have not yet upgraded to AndroidX (e.g. retrolambda), but others (e.g. Butterknife) have, we need to selectively disable jetify.

Disabling Jefity for the specified libs has been [implemented in an alpha release (or unreleased beta) of Android Gradle Plugin 4 days ago](https://issuetracker.google.com/issues/119135578#comment5).
I strongly discourage from depending on Google's internal experimental builds in our prod unless we do Dex hardering tests targered for E8 (which is Google's “recent” transcompiler from java bytecode to Dalvik/Art). This is out of the scope of this projet.

Another solution (very time-consuming and over-industrialized for such a small project) would be to break or dependencies in two dependencies: One that builds with Jetifier on and another Off and then use them as dependencies in our projet. Upside: During refactoring we could break down our app into components and develop them separatedly (and thus testing separatedly). This would speed up travis (since business logic code wouldn't need to run instrumentation code, but also require training for the rest of the team.

Technical information:
- lombok-ast can't be repackaged by Jetify (Duplicate License file when creating the zip (???))
- Butterknife uses reflection and in combination with lombok generates invalid code after being jettified (ClassNotFound...)
- Dexmaker intetactions with jetifier need to be verified, though we should be find since we don't do manipulations on graphical elements.
